### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by gandradep**
- register and log in, so that the data is private to them.
- introduce new transactions associated with a category.
- see the money spent on each category.
- Video [link](https://www.loom.com/share/9cccc3006df44835be0be52ede7177a6):
  - description of project
  - demo of the project features
  - highlight some interesting piece of code.

### CODE REVIEW

File: app/views/categories/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  <%= link_to category_path(item), class: 'text-decoration-none' do%>
+    <div class="row align-items-center m-1 p-1 border border-secondary">
+      <%= image_tag(item.icon_url, class: 'img-fluid col-4')%>
+      <p class= "col-4 text-center text-dark fs-5"><%= item.name%></p>
+      <p class= "col-4 text-center text-dark fs-5"><%=number_to_currency(item.transacctions.sum(:amount))%></p>
+    </div>
+  <%end%>
+<%end%>
+
+<%= link_to 'Add category', new_category_path, class:'my-3 btn btn-primary w-100 rounded-0' %>
```
Comment (TedLivist) [inline app/views/categories/index.html.erb]: - [x] Kindly improve your styling here. Your variation of the blue color in the header is inconsistent with the design. Kindly use the correct color variation to match the design. 

- [x] The "Add category" button should be fixed at the button based on the design. Kindly implement that as well.

- [x] Let's also try to maintain the design colors as much as possible. The button should have the greenish color of the design with a rectangular inner white border. Kindly implement that as well.

- [x] The button should also be somewhat larger than it currently is. Kindly fix that.

- [x] The font styling should also be maintained. Kindly style category cards and improve their fonts to match the design.

- [x] Kindly add the greyish background as we have in the design also.

  Your Implementation | Behance Design
  :---:|:---:
  urc | riew

File: app/views/categories/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  <%= f.label :icon_url, class: 'form-label' %>
+  <%= f.text_field :icon_url, class: 'form-control' %>
+  </div>
+
+  <%= f.submit "Save", class: 'mt-2' %>
+<%end%>
+<% if flash[:error] %>
+    <div><%= flash[:error] %></div>
+<% end %>
+</div>
```
Comment (TedLivist) [inline app/views/categories/new.html.erb]: - [x] The forms should mostly be the same as we have in the design. Kindly maintain the colors and the position of the submit button (top-right).

- [x] Kindly ensure you also maintain the styling of the input forms. As you can see they stretch from end to end without border-radius.

  Your Implementation | Behance Design
  :---:|:---:
  djww | siow


image

File: app/views/categories/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    <div class="d-flex justify-content-between m-1 p-2 border border-secondary">
+      <div>
+        <p class= "text-dark fs-5 m-0"><%= item.name%></p>
+        <p class= "text-dark fs-6"><%= item.created_at.strftime("%d %b %y")%></p>
+      </div>
+      <p class= "text-center text-dark fs-5"><%= number_to_currency(item.amount)%></p>
+    </div>
+<%end%>
+
+<%= link_to 'Add transaction', new_category_transacction_path(@category), class:'my-3 btn btn-primary w-100 rounded-0 fs-5' %>
```
Comment (TedLivist) [inline app/views/categories/show.html.erb]: - [x] Kindly improve the design of the show page based on the design. You can follow the suggestions listed in the related comment above.

  Your Implementation | Behance Design
  :---:|:---:
  bhw | riew

File: app/views/transacctions/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      <%= item.name%><br>
+    <%end%>
+  </div>
+
+  <%= f.submit "Save", class: 'mt-2' %>
+<%end%>
+<% if flash[:error] %>
+    <div><%= flash[:error] %></div>
+<% end %>
+</div>
```
Comment (TedLivist) [inline app/views/transacctions/new.html.erb]: - [x] Kindly improve the design of the form based on the required design.

  ofew

File: app/views/categories/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<%= render "shared/nav_bar"%>
+<%= render "shared/messages"%>
+<div class="d-flex justify-content-between m-1 p-1">
+  <p class= "fs-5"><%=@category.name%> category expenses:</p>
+  <p class= "fs-5"><%= number_to_currency(@transacctions.sum(:amount))%></p>
+</div>
+
+<% @transacctions.each do |item|%>
```
Comment (TedLivist) [inline app/views/categories/show.html.erb]: - [x] The transactions should be ordered in the most recent order. When I add a transaction it should come first on the list, not last. Kindly fix that.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @gandradep,

Good job so far!
✔️ Nicely written and passing tests
✔️ Implementation of UML diagram
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gandradep** [comment]
@TedLivist thanks for the review. The app looks nicer with the suggested changes.

**Review by AnselemOdims — APPROVED** [review]
### Hi @gandradep 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**